src/js/components/sidebar.jsx

```diff
@@ -140,7 +174,18 @@ export default class SideBar extends React.Component {
       return;
     }
 
-    this.setState(Object.assign({}, this.state, { activeTabId: tabId }));
+    // select by click feature
+    const state = this.state;
```

Comment:
This is not necessarily required, but you could do:

```javascript
const { selectByClick, tabsById } = this.state;
```

and use those instead of `state.{foo}`, but its up to you.